pkg/kubelet/api/v1alpha1/runtime/api.proto

```diff
@@ -358,9 +369,21 @@ message LinuxUser {
     repeated int64 additional_gids = 3;
 }
 
-message ContainerConfig {
-    // Name of the container.
+// ContainerMetadata holds all necessary information for building unique
+// container name.
+message ContainerMetadata {
+    // The name of the container. Same as the container name in the PodSpec.
     optional string name = 1;
+    // The attempt number of creating the container.
+    optional uint32 attempt = 2;
+}
+
+// ContainerConfig holds all the required and optional fields for creating a
+// container.
+message ContainerConfig {
+    // The metadata of the container. The container runtime should build a unique
+    // container name based on this.
```

Comment:
more wording nitting. Technically a name need not be created with this. I think it's more "This information will uniquely identify a container, and a runtime should leverage this to ensure correct operation. A runtime may also use this information to improve UX, such as by constructing a readable name".

Again, we can leave it if you'd like